codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessage.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+/**
+ * RedisMessage is base interface for codec-redis.
+ */
+public interface RedisMessage {
+
+    /**
+     * Returns the {@link RedisMessageType}.
+     *
+     * @return the message type.
+     */
+    RedisMessageType type();
```

Comment:
@mp911de - I agree it would nice no to have to do the `instanceof` block, but we are currently dependent on internals of the type and must cast anyways. The code you linked to appears to provide a helper class `CommandOutput` which allows the construction of the types based on pieces put together in the encoder. If we wanted to remove the requirement that the encoder has type knowledge we could but an `encode(List<Object>)` method on the `RedisMessage` interface.